pkg/networkutils/network.go

```diff
@@ -451,10 +452,98 @@ func (n *linuxNetwork) SetupHostNetwork(vpcCIDRs []string, primaryMAC string, pr
 		},
 	})
 
+	log.Debugf("iptableRules: %v", iptableRules)
+	return iptableRules, nil
+}
+
+func (n *linuxNetwork) buildIptablesConnmarkRules(vpcCIDRs []string, ipt iptablesIface) ([]iptablesRule, error) {
+	var allCIDRs []string
+	allCIDRs = append(allCIDRs, vpcCIDRs...)
+	allCIDRs = append(allCIDRs, n.excludeSNATCIDRs...)
+	excludeCIDRs := sets.NewString(n.excludeSNATCIDRs...)
+
+	log.Debugf("Total CIDRs to exempt from connmark rules - %d", len(allCIDRs))
+	var chains []string
+	for i := 0; i <= len(allCIDRs); i++ {
+		chain := fmt.Sprintf("AWS-CONNMARK-CHAIN-%d", i)
+		log.Debugf("Setup Host Network: iptables -N %s -t nat", chain)
+		if err := ipt.NewChain("nat", chain); err != nil && !containChainExistErr(err) {
+			log.Errorf("ipt.NewChain error for chain [%s]: %v", chain, err)
+			return []iptablesRule{}, errors.Wrapf(err, "host network setup: failed to add chain")
+		}
+		chains = append(chains, chain)
+	}
+
+	var iptableRules []iptablesRule
+	log.Debugf("Setup Host Network: iptables -t nat -A PREROUTING -i eni+ -m comment --comment \"AWS, outbound connections\" -m state --state NEW -j AWS-CONNMARK-CHAIN-0")
+	iptableRules = append(iptableRules, iptablesRule{
+		name:        "connmark rule for non-VPC outbound traffic",
+		shouldExist: !n.useExternalSNAT,
+		table:       "nat",
+		chain:       "PREROUTING",
+		rule: []string{
+			"-i", "eni+", "-m", "comment", "--comment", "AWS, outbound connections",
```

Comment:
If AWS_VPC_K8S_CNI_VETHPREFIX is configured to something else than eni, this doesn't work. Need to make it based on the veth prefix configuration.

Reply:
fixed in new commit.